Vates/VatesAPI/src/vtkMDHexFactory.cpp

```diff
@@ -268,6 +266,44 @@ vtkMDHexFactory::create(ProgressAction &progressUpdating) const {
 }
 
 /*
+ * Get the next highest bin boundary
+ */
+coord_t
+vtkMDHexFactory::getNextBinBoundary(IMDEventWorkspace_sptr imdws) const {
+  auto t_dim = imdws->getTDimension();
+  coord_t bin_width = t_dim->getBinWidth();
+  coord_t dim_min = t_dim->getMinimum();
+  return roundUp(coord_t(m_time) - dim_min, bin_width) + dim_min;
+}
+
+/*
+ * Get the previous bin boundary, or the current one if m_time is on a boundary
+ */
+coord_t
+vtkMDHexFactory::getPreviousBinBoundary(IMDEventWorkspace_sptr imdws) const {
+  auto t_dim = imdws->getTDimension();
+  coord_t bin_width = t_dim->getBinWidth();
+  coord_t dim_min = t_dim->getMinimum();
+  return roundDown(coord_t(m_time) - dim_min, bin_width) + dim_min;
```

Comment:
I think `static_cast<coord_t>(m_time)` is preferred over c-style casts. Also you can use `auto` up there.